# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My lane:

I chose Lane 2: Refresh / Content Opportunity Scoring.

Organizations often manage large inventories of content but have limited resources available for review. It is rarely practical to review every page manually. A ranking system can help prioritize which pages deserve attention first based on observable search and engagement signals.

This lane directly supports an operational decision: determining which pages should be reviewed first for refresh, expansion, monitoring, or optimization. Machine learning may help identify patterns across multiple signals that are difficult to capture with simple rules alone.

The starter dataset and baseline workflow are specifically designed around this problem, making it a strong starting point for a seven-week capstone project.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

## Search Question

Which content pages should be reviewed first for refresh or optimization based on observable search and engagement signals?

## Unit of Analysis

One content page.

## Output

A ranked content opportunity score.

## Decision

Which pages should be reviewed first by a content reviewer?

## Action

A reviewer may:

- Refresh outdated content
- Expand thin content
- Investigate declining content
- Improve engagement
- Monitor pages showing risk signals

## Cost of a Wrong Recommendation

### False Positive

A page is recommended even though it does not require attention. This wastes reviewer time and resources.

### False Negative

A page with meaningful opportunity is missed. This results in lost opportunities to improve traffic, engagement, or search visibility.

## Why Data and ML May Help

Content performance is influenced by multiple factors such as impressions, clicks, freshness, age, engagement, and search visibility. Machine learning may improve prioritization by detecting interactions among these signals that simple rules miss.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Unique content pages:", df["content_id"].nunique())
print("Declining pages:", (df["trend_direction"]=="down").sum())
print("Declining percentage:",
      round((df["trend_direction"]=="down").mean()*100,2), "%")

print("\nAverage impressions:",
      round(df["impressions_90d"].mean(),2))

print("Average sessions:",
      round(df["sessions_90d"].mean(),2))

print("Average CTR:",
      round(df["ctr"].mean(),4))

Rows: 30000
Unique content pages: 30000
Declining pages: 16262
Declining percentage: 54.21 %

Average impressions: 5200.37
Average sessions: 37.07
Average CTR: 0.5107


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Quick Look at the Data

The starter dataset contains 30000 rows and 30000 unique content pages.

Approximately 16262 of pages are labeled as declining according to the starter label definition.

The average page receives 5200.37 impressions and 37.07 sessions over the 90-day observatiion period.

These numbers suggest that there is meaningful variation in content performance, making prioritization a useful problem to investigate.

## Self-check

Before you submit, confirm each line honestly:

- [y ] Every section above is filled — markdown thinking AND the code that backs it
- [y ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [y ] No client names, URLs, or private queries anywhere
- [ y] My claims use careful words: observed, measured, directional, decision-support
- [ y] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.